Upgrade to linux-io@1.0.1 for Node 10

If BeagleBone-IO is installed with [Node v10.0.0-nightly2018040653aaa55a3a](https://nodejs.org/download/nightly/v10.0.0-nightly2018040653aaa55a3a/) there are warnings.

Upgrading to linux-io@1.0.1 will resolve this issue.

```
$ node -v
v10.0.0-nightly2018040653aaa55a3a

$ npm i beaglebone-io

> epoll@1.0.2 install /home/pi/beaglebone-io/node_modules/epoll
> node-gyp rebuild

make: Entering directory '/home/pi/beaglebone-io/node_modules/epoll/build'
  CXX(target) Release/obj.target/epoll/src/epoll.o
In file included from ../src/epoll.cc:15:0:
../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’:
../../nan/nan.h:840:60: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, func, argc, argv);
                                                            ^
In file included from ../src/epoll.cc:12:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:172:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/epoll.cc:15:0:
../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’:
../../nan/nan.h:855:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, symbol, argc, argv);
                                                              ^
In file included from ../src/epoll.cc:12:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:165:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/epoll.cc:15:0:
../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’:
../../nan/nan.h:870:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, method, argc, argv);
                                                              ^
In file included from ../src/epoll.cc:12:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:158:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/epoll.cc:15:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const’:
../../nan/nan.h:1484:5: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
     ));
     ^
In file included from ../src/epoll.cc:12:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:172:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
  SOLINK_MODULE(target) Release/obj.target/epoll.node
  COPY Release/epoll.node
make: Leaving directory '/home/pi/beaglebone-io/node_modules/epoll/build'

> i2c-bus@1.2.5 install /home/pi/beaglebone-io/node_modules/i2c-bus
> node-gyp rebuild

make: Entering directory '/home/pi/beaglebone-io/node_modules/i2c-bus/build'
  CXX(target) Release/obj.target/i2c/src/i2c.o
In file included from ../src/i2c.cc:2:0:
../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’:
../../nan/nan.h:840:60: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, func, argc, argv);
                                                            ^
In file included from ../src/i2c.cc:1:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:172:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/i2c.cc:2:0:
../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’:
../../nan/nan.h:855:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, symbol, argc, argv);
                                                              ^
In file included from ../src/i2c.cc:1:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:165:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/i2c.cc:2:0:
../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’:
../../nan/nan.h:870:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, method, argc, argv);
                                                              ^
In file included from ../src/i2c.cc:1:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:158:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
In file included from ../src/i2c.cc:2:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const’:
../../nan/nan.h:1484:5: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
     ));
     ^
In file included from ../src/i2c.cc:1:0:
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:172:50: note: declared here
                 NODE_EXTERN v8::Local<v8::Value> MakeCallback(
                                                  ^
/home/pi/.node-gyp/10.0.0-nightly2018040653aaa55a3a/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
     __attribute__((deprecated(message))) declarator
                                          ^~~~~~~~~~
  SOLINK_MODULE(target) Release/obj.target/i2c.node
  COPY Release/i2c.node
make: Leaving directory '/home/pi/beaglebone-io/node_modules/i2c-bus/build'
npm WARN saveError ENOENT: no such file or directory, open '/home/pi/beaglebone-io/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/home/pi/beaglebone-io/package.json'
npm WARN beaglebone-io No description
npm WARN beaglebone-io No repository field.
npm WARN beaglebone-io No README data
npm WARN beaglebone-io No license field.
npm WARN You are using a pre-release version of node and things may not work as expected

+ beaglebone-io@3.0.0
added 21 packages in 36.755s
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,11 +1,7 @@
 language: node_js
 node_js:
-  - "0.10"
-  - "0.12"
   - "4"
-  - "5"
   - "6"
-  - "7"
   - "8"
   - "9"
 before_script:

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 [Johnny-Five](http://johnny-five.io) IO Plugin for the BeagleBone Black,
 BeagleBone Green Wireless and PocketBeagle.
 
-BeagleBone-IO supports Node.js version 0.10, 0.12, 4, 5, 6, 7, 8 and 9.
+BeagleBone-IO supports Node.js versions 4, 6, 8 and 9.
 
 ## Contents
 

--- a/README.md
+++ b/README.md
@@ -68,7 +68,22 @@ board.on('ready', function () {
 });
 ```
 
-Note, calling `analogRead`, `digitalRead` or `i2cRead` initiates a new **continuous** data reading process for a pin not a single read. Further details can be seen at https://github.com/rwaldron/io-plugins#analogreadpin-handler.
+Calling `analogRead`, `digitalRead` or `i2cRead` initiates a new
+**continuous** data reading process for a pin not a single read. Further
+details can be seen at the Johnny-Five
+[IO Plugins Specification](https://github.com/rwaldron/io-plugins#reading).
+
+Analog inputs will be disabled by default on some systems and must be enabled
+before usage. To enable analog inputs remove or comment out the
+following line in `/boot/uEnv.txt`
+```
+disable_uboot_overlay_adc=1
+```
+Attempting to use analog inputs that are not enabled will typically result in
+ENOENT error messages:
+```
+Error: ENOENT: no such file or directory, open '/sys/bus/iio/devices/iio:device0/in_voltage0_raw'
+```
 
 #### Using BeagleBone-IO with Johnny-Five
 

--- a/lib/pinmux.js
+++ b/lib/pinmux.js
@@ -7,7 +7,7 @@ var OCP_PATH = '/sys/devices/platform/ocp/',
 
 module.exports.state = function (pinId, state) {
   fs.writeFileSync(
-    OCP_PATH + '/ocp:' + pinId + '_pinmux/state',
+    OCP_PATH + 'ocp:' + pinId + '_pinmux/state',
     state,
     FS_OPTIONS
   );

--- a/lib/pwm-output.js
+++ b/lib/pwm-output.js
@@ -96,7 +96,8 @@ function PwmOutput(pinData, period) {
   pwmChannelPath = matches[0];
 
   // On kernel v4.11+ we wait for udev to set the permissions on the files
-  // used to control pwm enabling to be accessed without root privileges.
+  // used to control pwm enabling the files to be accessed without root
+  // privileges.
   waitForAccessPermission([
     pwmChannelPath + '/period',
     pwmChannelPath + '/enable',

--- a/package.json
+++ b/package.json
@@ -25,17 +25,17 @@
   },
   "homepage": "https://github.com/julianduque/beaglebone-io",
   "devDependencies": {
-    "grunt": "^1.0.1",
+    "grunt": "^1.0.2",
     "grunt-contrib-jshint": "^1.1.0",
     "grunt-contrib-nodeunit": "^1.0.0",
     "grunt-contrib-watch": "^1.0.0",
     "grunt-jsbeautifier": "^0.2.13",
-    "nodeunit": "^0.11.0",
-    "rewire": "^2.5.2",
-    "sinon": "^2.1.0"
+    "nodeunit": "^0.11.2",
+    "rewire": "^3.0.2",
+    "sinon": "^4.5.0"
   },
   "dependencies": {
-    "linux-io": "^0.9.2",
+    "linux-io": "^1.0.1",
     "glob": "^7.1.2"
   }
 }

--- a/package.json
+++ b/package.json
@@ -10,6 +10,9 @@
     "type": "git",
     "url": "https://github.com/julianduque/beaglebone-io"
   },
+  "engines": {
+    "node": ">=4.0.0"
+  },
   "keywords": [
     "firmata",
     "beaglebone",

